### ISSUE TITLE
Legal: The licence is invalid

### ISSUE THREAD

**Issue by Lamparter** [files-community/Files#16083]
**For the attention of @yaira2**

_TL;DR:_ the copyright on Files is invalid.
---

Hello,

While inspecting inspecting and experimenting with the codebase it has come into light that there is a significant oversight in the copyright notice of Files. Specifically, the notice is missing the first year of publication which is a fundamental requirement for copyright validity under international copyright law, especially in the United States and the United Kingdom.

The absence of the initial publication year could potentially lead into ambiguities regarding copyright on GitHub and the Microsoft Open Source agreement which states that if the licence is invalid every single contributor on a project will be in violation of the Visual Studio Community licence, of which I assume the contributors to this project, including Files organisation developers, use on a daily basis. 

To rectify this, the copyright notice must be amended to include the first year of the project's creation (2018). This will involve updating the notice in all files where it appears, particularly references across source code files and of course the licence in the root directory. The revised notice should read as follows: `Copyright (c) 2018-2024, Files Community` #14933

### Concerned code

This involves all code which references a licence year, and the licence in the root directory.

### Gains

_This issue would resolve copyright errors that could theoretically violate international copyright law rendering the copyright invalid._

### Requirements

- The copyright timespan must be updated.

**Comment by yaira2** [comment]
I’m not an expert, but from what I’ve found, it seems like including the year of first publication is optional. However, I don't see any harm in including the year, feel free to create a PR, and I’ll be happy to merge it when I get a chance!

**Comment by Lamparter** [comment]
> I’m not an expert, but from what I’ve found, it seems like including the year of first publication is optional. However, I don't see any harm in including the year. Feel free to create a PR, and I’ll be happy to merge it when I get a chance!

Ah yes, turns out after re-reading a bit closer the Berne convention (the law saying that timespan is optional) is internationally recognised which I didn't initially realise. I'll submit a PR though anyway, which you can decline or accept if you see fit 😁
